src/cls/IPM/Main.cls

```diff
@@ -889,20 +889,39 @@ ClassMethod Namespace(ByRef pCommandInfo) [ Internal ]
 /// Get list Namespace, example do ##class(%IPM.Main).GetListNamespace(.ns)
 ClassMethod GetListNamespace(Output list, pSearch As %String = "")
 {
+  Kill list
   Set list = 0
+
+  // Build pattern for matching outside of SQL. 
+  // Directly using `where Nsp LIKE ?` causes a bug described in https://github.com/intersystems/ipm/issues/579 
+  Set pSearch = $Zstrip(pSearch, "<>WC")
+  If pSearch = "" {
+    Set pSearch = "*"
+  }
+  Set pieces = $ListFromString(pSearch, "*")
+  Set transformedPieces = ""
+  Set ptr = 0
+  While $ListNext(pieces, ptr, val) {
+    If val '? .(1AN,1"%") { // namespace contain only numbers, letters and %, otherwise there's guaranteed no match
+      Return
+    }
+    Set transformedPieces = transformedPieces _ $ListBuild(1_$$$QUOTE(val))
+  }
+  Set pattern = $ListToString(transformedPieces, ".(1AN,1""%"")")
```

Comment:
Nit: Better to pattern match using $Match with regex as it is more performant (also regex is easier for anyone to understand instead of proprietary matching rules in objectscript)